boards/arm/nucleo_g431rb/nucleo_g431rb.yaml

```diff
@@ -0,0 +1,20 @@
+identifier: nucleo_g431rb
+name: NUCLEO-G431RB
+type: mcu
+arch: arm
+toolchain:
+  - zephyr
+  - gnuarmemb
+  - xtools
+ram: 32
+flash: 128
+supported:
+  - arduino_i2c
+  - nvs
```

Comment:
You can add arduino_spi and arduino_gpio

Reply:
Done